setup.py

```diff
@@ -121,6 +121,55 @@ def build_extensions(self):
 REQUIRES = [r.replace('[array]', '') for r in REQUIRES]
 
 
+compile_flags = ['-fopenmp']
+link_flags = ['-fopenmp']
+
+code = """#include <omp.h>
+int main(int argc, char** argv) { return(0); }"""
+
+
+class ConditionalOpenMP(build_ext):
+
+    def can_compile_link(self):
+
+        cc = self.compiler
+        fname = 'test.c'
+        cwd = os.getcwd()
+        tmpdir = tempfile.mkdtemp()
+
+        try:
+            os.chdir(tmpdir)
+            with open(fname, 'wt') as fobj:
+                fobj.write(code)
+            try:
+                objects = cc.compile([fname],
+                                     extra_postargs=compile_flags)
+            except CompileError:
+                return False
+            try:
+                # Link shared lib rather then executable to avoid
+                # http://bugs.python.org/issue4431 with MSVC 10+
+                cc.link_shared_lib(objects, "testlib",
+                                   extra_postargs=link_flags)
+            except (LinkError, TypeError):
+                return False
+        finally:
```

Comment:
FYI, this `try-finally` could we rewritten using `with tempfile.TemporaryDirectory() as tmpdir:` context manager.